src/lib/gadgets/rot.ts

```diff
@@ -0,0 +1,118 @@
+import { Field } from '../field.js';
+import { Provable } from '../provable.js';
+import { Fp } from '../../bindings/crypto/finite_field.js';
+import * as Gates from '../gates.js';
+
+export { rot, rotate };
+
+const MAX_BITS = 64 as const;
+
+function rot(field: Field, bits: number, direction: 'left' | 'right' = 'left') {
+  // Check that the rotation bits are in range
+  if (bits < 0 || bits > MAX_BITS) {
+    throw Error(`rot: expected bits to be between 0 and 64, got ${bits}`);
+  }
+
+  if (direction !== 'left' && direction !== 'right') {
+    throw Error(
+      `rot: expected direction to be 'left' or 'right', got ${direction}`
+    );
+  }
```

Comment:
nit: according to my philosophy, we don't do runtime checks for cases that are impossible according to the types. wdyt?

Reply:
I guessed this was implicit from the input type being either `'left'` or `'right'`, no?

Reply:
Fixed with https://github.com/o1-labs/o1js/pull/1182/commits/33aff5c6848aa2421b27a6bf21c70137271e8954

My original reasoning is that TypeScript stops you from passing in a different value, but you can still "force" it with JS. I think you're right, and the check was removed :+1: